src/settlement/dto/create-settlement.dto.ts

```diff
@@ -0,0 +1,107 @@
+import { SettlementStatusEnum, thirdPartyResponseStatusEnum } from '@prisma/client';
+import { Transform, TransformFnParams } from 'class-transformer';
+import {
+  IsDate,
+  IsEnum,
+  IsInt,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+} from 'class-validator';
+
+// createSettlementDto is used for creating the new settlement request
+export class CreateSettlementDto {
+  // Request ID associated with the request
+  @IsNotEmpty()
+  @IsInt()
+  requestId: number;
+
+  // User ID associated with the request.
+  @IsNotEmpty()
+  @IsInt()
+  userId: number;
+
+  // Status of the request, should be one of the valid SettlementStatusEnum values.
+  @IsNotEmpty()
+  @IsEnum(SettlementStatusEnum)
+  requestStatus: SettlementStatusEnum;
+
+  // Status of the request, should be one of the valid UserResponseStatusEnum values.
+  @IsNotEmpty()
+  @IsEnum(thirdPartyResponseStatusEnum)
+  thirdPartyResponseStatus: thirdPartyResponseStatusEnum;
+
+  // Optional content of the request for the transaction ID.
+  @IsOptional()
+  @IsNumber()
+  transactionID: number;
+
+  // Optional content of the request in JSON format.
+  @IsOptional()
+  // Use the @Transform decorator to apply a custom transformation to a property.
+  @Transform(({ value }: TransformFnParams) => {
+    try {
+      // Try to parse the input value as JSON to convert it into a JavaScript object.
+      return JSON.parse(value);
+    } catch (error) {
+      // If parsing fails (e.g., if value is not a valid JSON string), return the original value.
+      return value;
+    }
+  })
+  content?: object;
+
+  // Optional creation date of the request.
+  @IsDate()
+  @IsOptional()
+  createdAt?: Date;
+
+  // Optional update date of the request.
+  @IsDate()
+  @IsOptional()
+  updatedAt?: Date;
+}
+
+// SettlementStatusDto is used for updating the status of a request.
+export class SettlementStatusDto {
+  // New status for the request, should be one of the valid SettlementStatusEnum values.
+  @IsNotEmpty()
+  @IsEnum(SettlementStatusEnum)
+  requestStatus: SettlementStatusEnum;
+}
+
+// SettlementStatusDto is used for updating the status of a request.
+export class thirdPartyResponseStatusEnumDto {
+  // New status for the request, should be one of the valid UserResponseStatusEnum values.
+  @IsNotEmpty()
+  @IsEnum(thirdPartyResponseStatusEnum)
+  thirdPartyResponseStatus: thirdPartyResponseStatusEnum;
+}
```

Comment:
If there is no future use case for "SettlementStatusDto" and "thirdPartyResponseStatusEnumDto" please remove them as they are not used anywhere.